release_notes/ocp-4-12-release-notes.adoc

```diff
@@ -1267,6 +1267,26 @@ sourceStrategy:
+
+* Previously, there was a race condition that after a `routeSelector` update caused the ingress controller to clear the route status before the router deployment finished rolling out. This resulted in the status being incorrectly repopulated. With this update, route status is consistently cleared with a `routeSelector` update. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1962502[*BZ#1962502*])
+
+* Previously, a cluster that was upgraded from a version of {product-title} earlier than 4.8 could have `orphaned` `Route` objects. This is caused by earlier versions of {product-title} translating `Ingress` objects into `Route` objects irrespective of a given `Ingress` object’s indicated `IngressClass`. With this update, an administrator is made aware of `Ingress` objects that do not specify an `IngressClass` and `Route` objects that have been orphaned through an alert. The alert notifies administrators about any `orphaned` `Route` objects and any `Ingress` objects that do not specify any `IngressClass`. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1962502[*BZ#1962502*])
+
+* Previously, if a `configmap` that the router deployment depends on is not created, then the router deployment does not progress. With this update, the cluster Operator reports `ingress progressing=true` if the default ingress controller deployment is progressing. This results in users debugging issues with the ingress controller by using the command `oc get co`. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2066560[*BZ#2066560*])
+
+* Previously, CoreDNS was based on kubernetes 1.24 while {product-title} 4.12 is based on kubernetes 1.25. With this update, CoreDNS is v1.10.0, new features have been added, and {product-title} 4.12 uses kubernetes 1.25 APIs. (link: https://issues.redhat.com/browse/OCPBUGS-1731[*OCPBUGS-1731*])
+
+* Previously, routers used GO kubernetes libraries that were based on kubernetes 1.24 while {product-title} 4.12 is based on kubernetes 1.25. With this update, GO kubernetes API is v1.25.2, which aligns the router with {product-title} 4.12 that uses kubernetes 1.25 APIs. (link: https://issues.redhat.com/browse/OCPBUGS-1730[*OCPBUGS-1730*])
+
+* Previously, the Cluster Ingress Operator used GO kubernetes libraries that were based on kubernetes 1.24 while {product-title} 4.12 is based on kubernetes 1.25. With this update, GO kubernetes API is v1.25.2, which aligns the Cluster Ingress Operator with {product-title} 4.12 that uses kubernetes 1.25 APIs. (link: https://issues.redhat.com/browse/OCPBUGS-1554[*OCPBUGS-1554*])
+
+* Previously, the Cluster DNS Operator did not reconcile the `openshift-dns` namespace, which is a requirement for {product-title} 4.12 to have pod-security labels. With this update, the Cluster DNS Operator reconciles the `openshift-dns` namespace and the pod security labels are correct ensuring that pods startup with no problems.  (link: https://issues.redhat.com/browse/OCPBUGS-1549[*OCPBUGS-1549*])
```

Comment:
```suggestion
* Previously, the DNS Operator did not reconcile the `openshift-dns` namespace. Because {product-title} {product-version} requires the `openshift-dns` namespace to have pod-security labels, this caused the namespace to be missing those labels upon cluster update. Without the pod-security labels, the pods failed to start. With this update, the DNS Operator now reconciles the `openshift-dns` namespace, and the pod-security labels are now present. As a result, pods start up as expected. (link:https://issues.redhat.com/browse/OCPBUGS-1549[*OCPBUGS-1549*])
```